### PR TITLE
add the child specification

### PR DESCRIPTION
**Pull request by zekus**

### CODE REVIEW

File: lib/segment.ex [hunk 1]
```diff
@@ -10,6 +10,40 @@ defmodule Segment do
...
+    iex> Segment.child_spec([])
+    ** (KeyError) key :key not found in: []
+
+    iex> Segment.child_spec([key: "something", endpoint: "http://example.com"])
+    %{
+      id: Segment,
+      start: {Segment, :start_link, ["something", "http://example.com"]}
+    }
+
+    iex> Segment.child_spec([key: "something", endpoint: "http://example.com"])
```
Comment (prodis) [inline lib/segment.ex]: Duplicated?

File: lib/segment.ex [hunk 2]
```diff
@@ -10,6 +10,40 @@ defmodule Segment do
...
+    %{
+      id: Segment,
+      start: {Segment, :start_link, ["something", "http://example.com"]}
+    }
+
+  """
+  def child_spec(arg) do
+    %{
+      id: Segment,
+      start: {Segment, :start_link, [Keyword.fetch!(arg, :key), Keyword.get(arg, :endpoint)]}
```
Comment (prodis) [inline lib/segment.ex]: From what I understood `Keyword.fetch!/2` is used for required parameters and `Keyword.get/2` for optional parameters.

I'd suggest to move to outside the map the make more clear which one is required and optional. And also the make easier to find the error in the error stack when the `key` is not supplied. 

For example:
```elixir
key = Keyword.fetch!(arg, :key)
endpoint = Keyword.get(arg, :endpoint)

%{
  id: Segment,
  start: {Segment, :start_link, [key, endpoint]}
}
```

or 

```elixir
params = [
  Keyword.fetch!(arg, :key),
  Keyword.get(arg, :endpoint)
]

%{
  id: Segment,
  start: {Segment, :start_link, params}
}
```

**Review by prodis — APPROVED** [review]
LGTM

I left some suggestion though.

File: lib/segment.ex (hunk 1 above)
Comment (zekus) [reply lib/segment.ex]: 🤦‍♂️

**Pull request merged**